tests/std/tests/P0896R4_common_iterator/test.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include <cassert>
+#include <concepts>
+#include <iterator>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+struct instantiator {
+    template <input_or_output_iterator Iter>
+    static constexpr void call() {
+        if constexpr (copyable<Iter>) {
+            using Sen    = test::sentinel<iter_value_t<Iter>>;
+            using Cit    = common_iterator<Iter, Sen>;
+            int input[3] = {1, 2, 3};
+
+            // [common.iter.types]
+            using iconcept = typename iterator_traits<Cit>::iterator_concept;
+            if constexpr (forward_iterator<Iter>) {
+                STATIC_ASSERT(same_as<iconcept, forward_iterator_tag>);
+            } else {
+                STATIC_ASSERT(same_as<typename iterator_traits<Cit>::iterator_concept, input_iterator_tag>);
+            }
+
+            using icat = typename iterator_traits<Cit>::iterator_category;
+            if constexpr (derived_from<icat, forward_iterator_tag>) {
+                STATIC_ASSERT(same_as<icat, forward_iterator_tag>);
+            } else {
+                STATIC_ASSERT(same_as<icat, input_iterator_tag>);
+            }
+
+            using ipointer = typename iterator_traits<Cit>::pointer;
+            if constexpr (_Has_op_arrow<Iter>) {
+                STATIC_ASSERT(same_as<ipointer, decltype(declval<const Iter&>().operator->())>);
+            } else {
+                STATIC_ASSERT(same_as<ipointer, void>);
+            }
+
+            { // [counted.iter.const]
+                [[maybe_unused]] Cit defaultConstructed{};
+
+                [[maybe_unused]] Cit iterConstructed{Iter{input}};
+                [[maybe_unused]] Cit sentinelConstructed(Sen{});
+            }
+
+            { // [counted.iter.access]
+                Cit iter{Iter{input}};
+                assert(*iter == 1);
+
+                const Cit constIter{Iter{input}};
+                assert(*constIter == 1);
+            }
+
+            { // [counted.iter.nav]
+                Cit iter{Iter{input}};
+                ++iter;
+                assert(*iter == 2);
+
+                assert(*iter++ == 2);
+                assert(*iter == 3);
+            }
+
+            { // [counted.iter.cmp]
+                // Compare iterator / iterator
+                assert(Cit{Iter{input}} == Cit{Iter{input}});
+                assert(Cit{Iter{input}} != Cit{Iter{input + 1}});
+
+                // Compare iterator / sentinel
+                assert(Cit{Iter{input}} == Cit{Sen{input}});
+                assert(Cit{Sen{input}} != Cit{Iter{input + 1}});
+
+                // Compare sentinel / sentinel
+                assert(Cit{Sen{input}} == Cit{Sen{input}});
+                assert(Cit{Sen{input}} == Cit{Sen{input + 1}});
+            }
+
+            { // [common.iter.cust]
+                if constexpr (input_iterator<Iter>) { // iter_move
+                    Cit iter1{Iter{input}};
+
+                    const auto iter2 = ranges::iter_move(iter1);

```

Comment:
`iter2` is a very misleading name here. How about:
```suggestion
                    const same_as<int> auto element1 = ranges::iter_move(iter1);
```

Reply:
Ups,

I expanded this to `const same_as<iter_value_t<Iter>> auto element1 = ranges::iter_move(iter1);` in case someone wants to change the type some time later